### PR TITLE
Use podman for not root docker in test

### PR DESCRIPTION
**Pull request by namse**
Docker basically run as root, so if we mount directory to docker, the files docker made has root permission. So we cannot remove it without sudo.
It happened a lot because we pass cargo_home and source code to docker on `namui-test`.

To avoid that problem, I set ~rootless~ `podman` which use host's user directly.

After this, we don't have to make files chmod 777 on exit of test.

### CODE REVIEW

**Comment by namse** [comment]
@bigfoodK Please test installation in your environment before review.

**Review by bigfoodK — APPROVED** [review]
## I tested
1. Clear all cargo's targets
2. Run `install.sh`
3. Run `namui test` in luda-adventure

This pr worked for me.

FYI
If this causes problem, we can try [this docker option](https://docs.docker.com/engine/reference/run/#user) later.

File: namui-cli/scripts/install.sh [hunk 1]
```diff
@@ -184,4 +189,69 @@ function install_dot_env_file() {
...
+        sudo apt-get install -y dbus-user-session docker-ce-rootless-extras
+    fi
+
+    echo "If you have a problem, please reinstall docker like this doc, https://docs.docker.com/engine/install/ubuntu/"
+    echo "Setting docker as rootress mode..."
+
+    if [ ! -f ~/.config/systemd/user/docker.service ]; then
+        dockerd-rootless-setuptool.sh install --force
+
+        export PATH=/usr/bin:$PATH
```
[the remark this answers is not in the document]
Comment (bigfoodK) [reply namui-cli/scripts/install.sh:237]: Why should we add `/usr/bin` into path twice?

[the remark this answers is not in the document]
Comment (namse) [reply namui-cli/scripts/install.sh:237]: I think it's the problem that you have cargo in /usr/bin/cargo. I think don't have it (i guess???)

**Comment by namse** [comment]
> ## I tested
> 
> 1. Clear all cargo's targets
> 
> 2. Run `install.sh`
> 
> 3. Run `namui test` in luda-adventure
> 
> 
> 
> This pr worked for me.
> 
> 
> 
> FYI
> 
> If this causes problem, we can try [this docker option](https://docs.docker.com/engine/reference/run/#user) later.

Let me teat it. give me time

**Comment by namse** [comment]
Actually I failed to use `USER` before but I think I did something in wrong at that time. This time I think I fixed it.

@bigfoodK Please test it without rootless context. Maybe you have to call `docker context use default` and run docker daemon again.

**Comment by namse** [comment]
I was able to remove cargo cache in my ubuntu host after run `namui test`.

**Review by bigfoodK — COMMENTED** [review]
It didn't worked for me.

I tested with
- Re-install docker and use default context
- Remove targets and `~/.cargo/registry`
- Remove docker images
- Run namui-cli install script
- Check see if registry directory is removeable without sudo

`registry/index/*` is created with root. but strangley `registry/cache/*` is created without root.
Is my test procedure wrong?

**Comment by bigfoodK** [comment]
+ps `luda-adventure/target` was also created with root.

**Comment by namse** [comment]
@bigfoodK 
Ah, you have to build docker image in local. 

```bash
cd docker-images/namui-test-host/
docker build -t ghcr.io/namseent/namui-test-host -f ./linux.Dockerfile .
```

And run test please.

**Comment by bigfoodK** [comment]
Ah! good. it works

**Review by bigfoodK — APPROVED** [review]

**Comment by namse** [comment]
I was wrong that, I didn't changed binded directory path so it installed in local of docker.
I'll install rustup inside of docker for every user, using https://github.com/rust-lang/rustup/issues/1085,
Then I will run test-docker with `-user $(whoami)`.

**Comment by namse** [comment]
Please test by just install namui-cli and run the test without anything.

**Comment by namse** [comment]
Shit I forgot that you approved this PR :)

**Pull request merged**